src/JsonReader.cs

```diff
@@ -548,127 +570,162 @@ sealed class NonProperty : IJsonProperty<Unit, JsonReadResult<Unit>>
+                            else if (TrySetPropertyIndex(3,  property3,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(4,  property4,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(5,  property5,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(6,  property6,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(7,  property7,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(8,  property8,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(9,  property9,  ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(10, property10, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(11, property11, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(12, property12, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(13, property13, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(14, property14, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(15, property15, ref reader, ref values)) { }
+                            else if (TrySetPropertyIndex(16, property16, ref reader, ref values)) { }
+
+                            if (!reader.Read())
+                                return reader.Suspend((sm, values));
+
+                            sm.OnPropertyNameRead();
 
-                    switch (property.Reader.TryRead(ref reader))
-                    {
-                        case { Incomplete: true }:
-                            throw PartialJsonNotSupportedException();
-                        case { Error: { } err }:
-                            error = err;
-                            return false;
-                        case { Value: var val }:
-                            value = (true, val);
-                            return true;
+                            break;
+
+                        case ObjectReadStateMachine.ReadResult.PropertyValue:
+                            static JsonReadResult<TResult>? ReadPropertyValue<TValue>(IJsonProperty<TValue, JsonReadResult<TValue>> property,
+                                                                                      ref Utf8JsonReader reader,
+                                                                                      ref (bool, TValue) value,
+                                                                                      ref ObjectReadStateMachine sm,
+                                                                                      ref ObjectValueState<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16> values)
```

Comment:
This argument is never modified so it could be passed with the `in` modifier as opposed to `ref`. The `in` will pass by-reference but not allow modifications, which is more accurate to how the argument is used by the function. If you take the [earlier suggestion of making the function non-static](https://github.com/atifaziz/Jacob/pull/21/files#r1053120682) then you don't even need to pass this.

Reply:
Done in 30fcfb09a68344c238ad2c0680f35bdc3192371a